### ISSUE TITLE
Deploy the app on GitHub pages

### ISSUE THREAD

**Issue by surchs** [neurobagel/old-query-tool#4]
We want

- a static version of the app deployed publicly
- a https enabled neurobagel URL to reach the app at
- an automatic process that deploys the public app based on the most recent version of the main branch

- [x] Enable static build process for deployment in `nuxt.config.js`
- [x] Add `dotenv` nuxt module to buildmodules in `nuxt.config.js`
- [x] Add `API_QUERY_URL` as an repository variable
- [x] Set up deploy workflow following [nuxt documentation](https://nuxtjs.org/deployments/github-pages/#build-server-deployment)
- [x] Add neurobagel domain as the deployment domain
- [x] Update README.md

**Comment by surchs** [comment]
@rmanaem: I have just realized something: If we deploy this on a remote cloud service the deployment would probably be a lot easier, but you couldn't contact the API as long as it is only listening to McGill / VPN IPs. So I think we need to go with a deployment on the server here. What do you think?

**Comment by rmanaem** [comment]
>So I think we need to go with a deployment on the server here

Am I correct to assume you're referring to the MNI server?
If so then what would be the deployment plan be?

**Comment by surchs** [comment]
> Am I correct to assume you're referring to the MNI server?

yes

> If so then what would be the deployment plan be?

We could either pre-build the app and host the static files from the MNI server, or serve it with a node server. See here: https://nuxt.com/docs/getting-started/deployment

**More importantly though:** this shouldn't really be a problem / I was confused about where the request is coming from. In either scenario of deployment of the query tool, once you do send a request from the query tool it will originate from the users machine, not from the server that the query tool is being served from. The users machine can (and should) obviously be at McGill / in the VPN. In fact that is how you are currently locally testing the query tool against the API I think.

So bottom line: I think we can go with the cloud hosting service option and make our lives easier for the start.

**Comment by rmanaem** [comment]
Update: For the first release GitHub pages will be used for deployment.
